mocca-client/src/main/java/com/paypal/mocca/client/MoccaSerializer.java

```diff
@@ -233,15 +235,25 @@ private static boolean isPojo(Type type) {
         return !clazz.isPrimitive();
     }
 
+
+    private static boolean isEnum(Type type) {
+        return Enum.class.isAssignableFrom((Class<?>) type);
+    }
+
     /*
      * Returns a String containing the specification of a GraphQL variable name and its value,
      * as it is supposed to be written in the request payload
      */
     private String writeRequestVariable(String name, Object value, Type type) {
         final String prefix = name == null ? "" : name + ": ";
-        return type == String.class || type == Character.class || type == OffsetDateTime.class || type == Duration.class || type == UUID.class || type.getTypeName().equals("char") ?
-                prefix + "\\\"" + value.toString() + "\\\"" :
-                prefix + value.toString();
+        if(isEnum(type)) {
+           return prefix + value.toString();
```

Comment:
Why would we not print the variable name for **request** parameter enumerations?

Notice this method is specific to writing **request variables**, and enumerations parameters must have the parameter name. The **return** value is the one that doesn't need it.

I believe this change here is wrong, it should be like you did before, with an additional condition (for enum) added to the end of line 253.

I understand the tests, and GRE, are functional regardless of what I am saying in this comment. However, that to me just means there is another bug somewhere that is making this change functional in spite of that.

Reply:
Here prefix is the one which represents the parameter name. The reason I have to separate it out is because we want to get rid of quotes for enum value as the graphql layer does not like the enum value with quotes around it and it triggers the exception. It's similar to line 253 just that it's taking the variable name plus variable value without quotes.

Reply:
Understood, it makes sense. Thanks for the clarification.